src/uint/modular/constant_mod/macros.rs

```diff
@@ -14,8 +15,16 @@ macro_rules! impl_modulus {
                 $crate::Concat<Output = $crate::Uint<DLIMBS>>,
         {
             const LIMBS: usize = { $crate::nlimbs!(<$uint_type>::BITS) };
-            const MODULUS: $crate::Uint<{ $crate::nlimbs!(<$uint_type>::BITS) }> =
-                <$uint_type>::from_be_hex($value);
+            const MODULUS: $crate::Uint<{ $crate::nlimbs!(<$uint_type>::BITS) }> = {
+                let res = <$uint_type>::from_be_hex($value);
+
+                // Check that the modulus is odd
+                if (res.as_limbs()[0].0 & 1) == 0 {
```

Comment:
It's interesting that `clippy` doesn't complain here about the unnecessary parentheses. Is it because it's in a macro?

Reply:
Oh, possibly. I can remove it.

Reply:
They aren't considered unnecessary since they don't wrap the complete expression. clippy allows parentheses, even if their lack of placement forms an equivalent expr, so long as it splits a larger term.